app/spaces/[spaceid]/trustful/components/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { useState, type ReactNode } from 'react';
+import { CheckIcon, CopyIcon } from '@chakra-ui/icons';
+import { Tooltip } from '@chakra-ui/react';
+import { useAccount } from 'wagmi';
+
+interface CopyToClipboardButtonProps {
+  isUserAddress?: boolean;
+  isShare?: boolean;
+  showSvg?: boolean;
+  label?: string;
+  className?: string;
+  svgClassName?: string;
+  children?: ReactNode;
+}
+
+export const CopyToClipboardButton = ({
+  isUserAddress = false,
+  isShare = false,
+  showSvg = false,
+  className = '',
+  svgClassName = '',
+  label,
+  children,
+}: CopyToClipboardButtonProps) => {
+  const { address } = useAccount();
+
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [isVisible, setIsVisible] = useState(false);
+  // const prefixToGiveBadge = 'https://trustful.ing/give-badge';
```

Comment:
Can you please add comments on the commented sections, sharing why these are commented and if these are open action items?